jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/DataCloudConnection.java

```
@@ -222,80 +222,37 @@ public DataCloudResultSet getRowBasedResultSet(String queryId, long offset, long
-        return StreamingResultSet.of(queryId, executor, iterator);
+        return StreamingResultSet.of(queryId, iterator);
     }
 
     public DataCloudResultSet getChunkBasedResultSet(String queryId, long chunkId) throws DataCloudJDBCException {
         return getChunkBasedResultSet(queryId, chunkId, 1);
     }
 
     /**
-     * Checks if a given row range is available for a query.
-     * This method will wait until the row range specified by the other params is available in the given timeout.
-     *
+     * Checks if a given predicate is satisfied by the status of the specified query.
+     * This method will poll the query status and exit immediately when the predicate is satisfied.
+     * If the predicate is never satisfied, the method will continue trying until the provided timeout is reached.
+     * Note that it is up to the caller to ensure that the returned {@link QueryStatus} meets expectations.
      * @param queryId The identifier of the query to check
-     * @param offset The starting row offset.
-     * @param limit The quantity of rows relative to the offset to wait for
      * @param timeout The duration to wait for the engine have results produced.
-     * @param allowLessThan Whether to return early when the available rows is less than {@code offset + limit}
-     * @return The first status where the rows available meet the constraints defined by the parameters or the last status the server replied with.
+     * @param predicate The condition to check against the query status,
+     *                  you can use {@link QueryStatus#isExecutionFinished()}, {@link QueryStatus#allResultsProduced()}, and {@link QueryStatus#isResultProduced()},
+     *                  as well as the methods on {@link QueryStatus.Predicates} for more complex predicates involving pages of rows or chunks
+     * @return The first status that satisfies the predicate, or the last {@link QueryStatus} received before timeout.
```

Should we rather throw an exception if the query status condition couldn't be met? That way it would be super explicit that the method will only return if the condition was met